piff/writers.py

```diff
@@ -0,0 +1,257 @@
+            with FitsWriter._open(file_name) as writer:
+                yield writer
+            return
+        else:
+            raise NotImplementedError(f"No writer for extension {ext!r}.")
+
+    def write_struct(self, name, struct):
+        """Write a simple flat dictionary.
+
+        :param name:        Name used to save this struct.
+        :param struct:      A `dict` with `str` keys and `int`, `float`, `str`,
+                            `bool`, or `None` values.
+        """
+        raise NotImplementedError()
+
+    def write_table(self, name, table, metadata=None):
+        """Write a table via a numpy array.
+
+        :param name:      Name used to save this table.
+        :param metadata:  A `dict` of simple metadata to save with the table.
+                          Keys must be `str` (case may not be preserved) and
+                          values must be `int`, `float`, `str`, or `bool`.
+        """
+        raise NotImplementedError()
+
+    def write_array(self, name, array, metadata=None):
+        """Write a numpy array that does not have a structured dtype.
+
+        :param name:      Name used to save this array.
+        :param metadata:  A `dict` of simple metadata to save with the array.
+                          Keys must be `str` (case may not be preserved) and
+                          values must be `int`, `float`, `str`, or `bool`.
+        """
+        raise NotImplementedError()
+
+    def write_wcs_map(self, name, wcs_map, pointing):
+        """Write a regular a map of WCS objects and an optoinal pointing coord.
+
+        :param name:      Name used to save this struct in `Writer.write_array`.
+        :param wcs_map:   A `dict` mapping `int` chipnum to `galsim.BaseWCS`.
+        :param pointing:  A `galsim.CelestialCoord`, or `None`.
+        :param logger:    A logger object for logging debug info.
+        """
+        raise NotImplementedError()
+
+    def nested(self, name):
+        """Return a context manager that yields a new `Writer` that nests all
+        names it is given within this one.
+
+        :param name:     Base name for all objects written with the returned object.
+
+        :returns:     A context manager that yields a nested writer object.
+
+        It is implementation-defined whether the content written by the nested
+        writer is actually written immediately or only when the context manager
+        exits.
+        """
+        raise NotImplementedError()
+
+    def get_full_name(self, name):
+        """Return the full name of a data structure saved with the given name,
+        combining it with any base names added if this `Writer` was created by
+        the `nested` method.
+        """
+        raise NotImplementedError()
+
+
+class FitsWriter(Writer):
+    """A `Writer` implementation that writes to multiple FITS HDUs.
+
+    This reader is intended to writes files that would be readable Piff before
+    the `Reader` and `Writer` abstractions were added.
```

Comment:
This sentence has multiple errors.  But the following would suffice I think.
"This writer is used to write Piff data to a FITS file."